lib/chef/knife/ec2_server_create.rb

```diff
@@ -689,6 +689,8 @@ def bootstrap_common_params(bootstrap)
         bootstrap.config[:first_boot_attributes_from_file] = locate_config_value(:first_boot_attributes_from_file)
         bootstrap.config[:encrypted_data_bag_secret] = s3_secret || locate_config_value(:secret)
         bootstrap.config[:encrypted_data_bag_secret_file] = locate_config_value(:secret_file)
+        #cl_secret needs to be set as chef checks for this key
+        Chef::Config[:knife][:cl_secret] = s3_secret if locate_config_value(:s3_secret)
```

Comment:
This code is a little confusing. I'm not sure that it might need a little refactor. At least might consider something like this:

```
# retrieving the secret from S3 is unique to knife-ec2, so we need to set "command line secret" to the value fetched from S3
Chef::Knife::DataBagSecretOptions.set_cl_secret(s3_secret)
```

We should add a test either way.